### PR TITLE
RestRoute

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Depends on | #48

## Added
- Custom helper for creating the `Yiisoft\Router\Group` for each HTTP method's actions.
- `RestControllerTrait` like a lazy definition of actions
- Flexible detecting supported methods for the controller

Maybe `RestControllerTrait` should be removed after implementing flixible detecting methods? Need to think about it.

### CODE REVIEW

File: src/RestControllerTrait.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Yiisoft\Yii\Rest;
+
+use Psr\Http\Message\ResponseInterface;
+
+trait RestControllerTrait
```
Comment (samdark) [inline src/RestControllerTrait.php]: Is it useful for non-tests?

File: src/RestGroup.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+use Yiisoft\Router\RouteCollectorInterface;
+
+class RestGroup
+{
+    public static function create(string $prefix, string $controller): RouteCollectorInterface
+    {
+        $routes = self::createDefaultRoutes($prefix, $controller);
+        $group = Group::create($prefix, $routes);
+
+        // add middlewares here
```
Comment (samdark) [inline src/RestGroup.php]: What middlewares?

File: src/RestControllerTrait.php (hunk 1 above)
Comment (xepozz) [reply src/RestControllerTrait.php]: hmm...may be yes when IDE will suggest you already defined method signature, eg. you typed `ge` and IDE suggests you
 ```php
public function get(): ResponseInterface
{
    // ...
}
```

File: src/RestGroup.php (hunk 2 above)
Comment (xepozz) [reply src/RestGroup.php]: I want to make middleware that will convert any variables to the `ResponseInterface`

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**